tiledb/sm/c_api/tiledb.h

```diff
@@ -4493,6 +4534,99 @@ TILEDB_EXPORT int32_t tiledb_query_get_fragment_timestamp_range(
     uint64_t* t1,
     uint64_t* t2);
 
+/* ****************************** */
+/*          QUERY CONDITION       */
+/* ****************************** */
+
+/**
+ * Allocates a TileDB query condition object.
```

Comment:
Perhaps add a comment that this applies only to attributes. Conditions on dimensions are effectively the (single- or multi-range) subarray slices added to the query object.